test/kinesis_helper/test_client.rb

```diff
@@ -46,21 +72,31 @@ def teardown
   end
 
   def test_instance_profile
+    Aws.shared_config.fresh
     setup_instance_profile
-    credentials = @object.client.config.credentials
+    credentials = MockClientInstanceProfile.new.client.config.credentials
     assert_equal 'akid',          credentials.credentials.access_key_id
     assert_equal 'secret',        credentials.credentials.secret_access_key
     assert_equal 'session-token', credentials.credentials.session_token
   end
 
   def test_instance_profile_refresh
+    Aws.shared_config.fresh
     setup_instance_profile(Time.now.utc + 299)
-    credentials = @object.client.config.credentials
+    credentials = MockClientInstanceProfile.new.client.config.credentials
     assert_equal 'akid-2',          credentials.credentials.access_key_id
     assert_equal 'secret-2',        credentials.credentials.secret_access_key
     assert_equal 'session-token-2', credentials.credentials.session_token
   end
 
+  def test_process_credentials
+    omit_if(Gem::Version.new(Aws::CORE_GEM_VERSION) < Gem::Version.new('3.24.0'))
```

Comment:
We need the same validation on `client.rb` as well. If the core SDK version is too old, it should throw configuration error to the customer.